apps/wallet/src/background/accounts/zklogin/ZkLoginAccount.ts

```diff
@@ -91,6 +94,19 @@ export function isZkLoginAccountSerializedUI(
 	return account.type === 'zkLogin';
 }
 
+async function hasTransactionHistory(address: string): Promise<boolean> {
+	const activeNetwork = await networkEnv.getActiveNetwork();
+	if (activeNetwork.env !== API_ENV.mainnet) {
+		return false;
+	}
+
+	const rpc = getSuiClient(activeNetwork);
+	const txns = await fetchTransactionsByAddress(rpc, address);
```

Comment:
Honestly I wouldn't bother doing this generalization, it's just two queries, not worth doing this for.

Reply:
This also will be weird with graphQL so I'd revert this generalization for sure